src/main/host/descriptor/mod.rs

```diff
@@ -32,7 +35,160 @@ impl<T> SyncSendPointer<T> {
     }
 }
 
-pub enum PosixFile {}
+#[derive(PartialEq)]
+pub enum SyscallReturn {
+    Success(i32),
+    Error(nix::errno::Errno),
+}
+
+bitflags::bitflags! {
+    // file modes: https://github.com/torvalds/linux/blob/master/include/linux/fs.h#L111
+    pub struct FileMode: u32 {
+        const READ = 0b00000001;
```

Comment:
As far as I can tell these are internal values and aren't exposed outside of the kernel, so they aren't in `libc`. I think it's okay if we use our own values here since the kernel doesn't take these values from userspace, so they should never get mixed up.

This did lead me to realize I wasn't handling flags very well, so I added a new commit `ba7f1fb6` to organize it better.

Reply:
Sounds good - a comment with some of this explanation would be helpful. With just the bare link it's not clear that these are "kernel private" and that we're just using as "inspiration" rather than actually needing to keep the values in sync.

Reply:
Nevermind; already addressed in 85bbb753fed6cb6ca4c56c9ac4b47e15d9acd7b9. (I wish GH would show comment threads alongside later versions of the code...)

Reply:
(Ah it does show them if I look at the "files changed" view instead of the "conversation" view)